src/parser/type_checking.c

```diff
@@ -12982,28 +12957,193 @@ pt_character_length_for_node (PT_NODE * node, const PT_TYPE_ENUM coerce_type)
   return precision;
 }
 
+namespace Func
+{
+  #define X(id, ...) #id,
+  const char* type_str[] = {
+    #include "func_type.x"
+  };
+  #undef X
+
+  #define X(id, signatures) signatures,
+  std::vector<func_signature>* types[] = {
+    #include "func_type.x"
+  };
+  #undef X
+
+  bool cmp_types_normal(const pt_arg_type& type, pt_type_enum type_enum)
+  {
+    return (type.type == pt_arg_type::NORMAL && type.val.type == type_enum);
```

Comment:
do we need checking the type.type == pt_arg_type::NORMAL ?
If so, maybe should have similar condition on cmp_types_generic.

Reply:
cmp_types_normal() only compares two pt_type_enum values and this makes sense only for pt_arg_type::NORMAL

cmp_types_generic() calls pt_are_equivalent_types() and this one can handle both NORMAL & GENERIC (INDEX makes no sense, I'll put an assert there for this case)